eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/test/Providers/ResourceCollectionClientProviderTests.cs

```diff
@@ -149,5 +151,53 @@ public void Verify_GetIfExistsAsyncOperationMethod()
             var exptected = Helpers.GetExpectedFromFile();
             Assert.AreEqual(exptected, bodyStatements);
         }
+
+        [TestCase]
+        public void Verify_CreateOrUpdateOperationMethod()
+        {
+            ResourceCollectionClientProvider resourceProvider = GetResourceCollectionClientProvider();
+            var methods = resourceProvider.Methods.Select(m => m.Signature.Name).ToArray();
+            Console.WriteLine($"Available methods: {string.Join(", ", methods)}");
+
+            MethodProvider createOrUpdateMethod = GetResourceCollectionClientProviderMethodByName("CreateOrUpdate");
+
+            // verify the method signature
+            var signature = createOrUpdateMethod.Signature;
+            Assert.AreEqual(MethodSignatureModifiers.Public | MethodSignatureModifiers.Virtual, signature.Modifiers);
+            Assert.AreEqual(3, signature.Parameters.Count);
+            Assert.AreEqual(typeof(string), signature.Parameters[0].Type.FrameworkType);
+            Assert.IsTrue(signature.Parameters[1].Type.Name.EndsWith("Data"));
+            Assert.AreEqual(typeof(CancellationToken), signature.Parameters[2].Type.FrameworkType);
+            Assert.AreEqual(typeof(Response<>), signature.ReturnType?.FrameworkType);
+
+            // verify the method body
+            var bodyStatements = createOrUpdateMethod.BodyStatements?.ToDisplayString();
+            Assert.NotNull(bodyStatements);
+            // TODO: Add expected output file for body verification
+            // var expected = Helpers.GetExpectedFromFile();
+            // Assert.AreEqual(expected, bodyStatements);
```

Comment:
Could we finish this?
You could find data files for other test cases under the `TestData` sub directory of this file, with the name of the corresponding test case